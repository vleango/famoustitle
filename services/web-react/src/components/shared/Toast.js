import React  from 'react';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

export const toastInProgress = (message) => {
    return toast(message, { autoClose: false });
};

export const toastSuccess = (message, toastID = null) => {
    if(toastID !== null) {
        toast.update(toastID, { render: message, type: toast.TYPE.SUCCESS, autoClose: 3000 });
    } else {
        toast.success(message);
    }
};

export const toastFail = (message, toastID = null) => {
    if(toastID !== null) {
        toast.update(toastID, { render: message, type: toast.TYPE.ERROR, autoClose: 3000 });
    } else {
        toast.error(message);
    }
};

export default () => {

    return (
        <ToastContainer position="top-center"
                        autoClose={5000}
                        hideProgressBar
                        newestOnTop={false}
                        closeOnClick
                        rtl={false}
                        pauseOnVisibilityChange
                        draggable
                        pauseOnHover />
    );
}
