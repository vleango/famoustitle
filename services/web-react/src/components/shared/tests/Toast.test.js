import React from 'react';
import { shallow } from 'enzyme';
import Toast from '../Toast';

let wrapper;

describe('Shared', () => {
    describe('Toast', () => {
        wrapper = shallow(<Toast />);
        it('should correctly render Toast', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

});
