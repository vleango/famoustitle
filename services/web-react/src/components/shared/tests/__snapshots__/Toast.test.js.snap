// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Toast should correctly render Toast 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoClose": 5000,
      "bodyClassName": null,
      "className": null,
      "closeButton": <CloseButton
        ariaLabel="close"
/>,
      "closeOnClick": true,
      "draggable": true,
      "draggablePercent": 80,
      "hideProgressBar": true,
      "newestOnTop": false,
      "pauseOnHover": true,
      "pauseOnVisibilityChange": true,
      "position": "top-center",
      "progressClassName": null,
      "rtl": false,
      "style": null,
      "toastClassName": null,
      "transition": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoClose": 5000,
        "bodyClassName": null,
        "className": null,
        "closeButton": <CloseButton
          ariaLabel="close"
/>,
        "closeOnClick": true,
        "draggable": true,
        "draggablePercent": 80,
        "hideProgressBar": true,
        "newestOnTop": false,
        "pauseOnHover": true,
        "pauseOnVisibilityChange": true,
        "position": "top-center",
        "progressClassName": null,
        "rtl": false,
        "style": null,
        "toastClassName": null,
        "transition": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
